core/closures.ml

```diff
@@ -373,11 +374,11 @@ struct
   let binder (x, (t, name, _)) = (x, (t, name, `Global))
   let fun_def (f, lam, z, location) = (binder f, lam, z, location)
   let binding = function
-    | `Let (x, body) -> `Let (binder x, body)
-    | `Fun def -> `Fun (fun_def def)
-    | `Rec defs -> `Rec (List.map fun_def defs)
-    | `Alien (x, n, language) -> `Alien (binder x, n, language)
-    | `Module _ -> failwith "unimplemented"
+    | Ir.Let (x, body) -> Ir.Let (binder x, body)
```

Comment:
Are there name clashes only on some constructors? I tried to avoid prefixing only some of the constructors, i.e. I would either prefix all or none. Will adding a type signature on `binding` allow to avoid quantifiers (probably not)? Also, I only see `Ir` being open in this module, so what do the constructor names clash with? Are there local opens?

Reply:
Agreed, this should be more consistent. I usually tried not to open entire modules globally inside of modules, but either qualify all constructors or open the necessary modules locally, similar to how you did it in the frontend. This spurious qualification is a result of me experimenting around, removed it.